### ISSUE TITLE
Add ability to set grpc-js MetadataOptions when using GrpcTransport

### ISSUE THREAD

**Issue by kskalski** [timostamm/protobuf-ts#259]
When grpc-js calls are made, it's possible to set [``Metadata``](https://grpc.github.io/grpc/node/grpc.Metadata.html), which besides key-value pairs, also contains a few pre-defined options set in constructor with [``MetadataOptions`` object](https://github.com/grpc/grpc-node/blob/7239c0e2a234db7e8226394c4ad84cd6059a6639/packages/grpc-js/src/metadata.ts#L71) ).

Currently the generated rpc clients in protobuf-ts take ``RpcOptions`` object, which contains ``meta: RpcMetadata``, but this option doesn't seem to support populating the same options as in ``MetadataOptions`` when call is made using ``GrpcTransport``.

I suppose ``GrpcMetadata`` object could be added to ``grpc-transport`` package, which would extend ``RpcMetadata`` and calls using grpc transport could translate them to fully populate ``Metadata``. I'm not sure if this translation could be done in a type-safe way - this would probably require baking in some extensibility into generic client.

**Comment by timostamm** [comment]
Yes, I agree with adding `metadataOptions?: MetadataOptions` to `GrpcCallOptions`. Thanks for opening the PR.

**Comment by kskalski** [comment]
This PR https://github.com/timostamm/protobuf-ts/pull/261 should allow the options to be passed by user

**Comment by Janaka-Steph** [comment]
Does `[grpcweb-transport](https://github.com/kskalski/protobuf-ts/tree/master/packages/grpcweb-transport)` is affected by the same issue?

**Comment by kskalski** [comment]
``grpcweb-transport`` doesn't use ``grpc-js``, so it doesn't consume its ``MetadataOptions``. One could imagine some of the functionalities from those options to be implemented in ``grpcweb-transport``, but that is an independent feature.
